app/entity-repositories/urlRepository.js

```diff
@@ -0,0 +1,47 @@
+const UrlFactory = require('../entities/url-entity/urlFactory');
+
+class UrlRepository {
+    constructor({ dbProvider }) {
+        this.dbProvider = dbProvider;
+        this.urlFactory = new UrlFactory();
```

Comment:
I've seen it before, but you dont need to construct those factories here. 
Someday we will learn basics of unit-testing, and u would see why this is a bad approach